signing_clients/apps.py

```diff
@@ -79,22 +77,43 @@ def ignore_certain_metainf_files(filename):
     return False
 
 
-def file_key(zinfo):
-    '''
-    Sort keys for xpi files
-    @param name: name of the file to generate the sort key from
-    '''
-    # Copied from xpisign.py's api.py and tweaked
-    name = zinfo.filename
+def file_key(filename):
+    """Sort keys for xpi files
+
+    The filenames in a manifest are ordered so that files not in a
+    directory come before files in any directory, ordered
+    alphabetically but ignoring case, with a few exceptions
+    (install.rdf, chrome.manifest, icon.png and icon64.png come at the
+    beginning; licenses come at the end).
+
+    This order does not appear to affect anything in any way, but it
+    looks nicer.
+    """
     prio = 4
-    if name == 'install.rdf':
+    if filename == 'install.rdf':
         prio = 1
-    elif name in ["chrome.manifest", "icon.png", "icon64.png"]:
+    elif filename in ["chrome.manifest", "icon.png", "icon64.png"]:
         prio = 2
-    elif name in ["MPL", "GPL", "LGPL", "COPYING", "LICENSE", "license.txt"]:
+    elif filename in ["MPL", "GPL", "LGPL",
+                      "COPYING", "LICENSE", "license.txt"]:
         prio = 5
-    parts = [prio] + list(os.path.split(name.lower()))
-    return "%d-%s-%s" % tuple(parts)
+    return (prio, os.path.split(filename.lower()))
+
+
+def manifest_header(type_name, version='1.0'):
+    """Returns a header, suitable for use in a manifest.
+
+    >>> manifest_header("signature")
+    "Signature-Version: 1.0"
+
+    :param type_name: The kind of manifest which needs a header:
+        "manifest", "signature".
+    :param version: The manifest version to encode in the header
+        (default: '1.0')
+    """
+    return u"%s-Version: %s" % (
+        type_name.title(),
+        version)
```

Comment:
Can we use string formatting like `u'{0}-Version'.format(type_name.title(), version)`, also I'm pretty sure it'll fit on one line :)